torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1850,6 +1850,85 @@ def sample_inputs_stack(op_info, device, dtype, requires_grad, **kwargs):
         for dim in range(-1, len(shape) - 1):
             yield SampleInput(tensors, args=(dim,))
 
+def error_inputs_cat(op_info, device, **kwargs):
+    # error inputs for unsupported operation
+    x = torch.rand((1, 3), device=device).expand((6, 3))
+    y = torch.rand((3, 3), device=device)
+    yield ErrorInput(SampleInput((y, y), kwargs={'out': x}),
+                     error_regex='unsupported operation')
+
+    # error inputs for empty tensors
+    yield ErrorInput(SampleInput([], kwargs={'dim': 1}),
+                     error_regex='non-empty list of Tensors')
+
+    # error inputs for different sizes
+    dtype = torch.float32
+    x = torch.randn((4, 3, 32, 32), dtype=dtype, device=device)
+    empty = torch.randn((4, 0, 31, 32), dtype=dtype, device=device)
+    yield ErrorInput(SampleInput((x, empty), kwargs={'dim': 1}),
+                     error_regex='Sizes of tensors must match except in dimension')
+    yield ErrorInput(SampleInput((empty, x), kwargs={'dim': 1}),
+                     error_regex='Sizes of tensors must match except in dimension')
+
+    # error inputs for different dimensions
+    empty = torch.randn((4, 0), dtype=dtype, device=device)
+    yield ErrorInput(SampleInput((x, empty), kwargs={'dim': 1}),
+                     error_regex='Tensors must have same number of dimensions')
+    yield ErrorInput(SampleInput((empty, x), kwargs={'dim': 1}),
+                     error_regex='Tensors must have same number of dimensions')
+
+    # error inputs for same memory locations
+    x = torch.zeros((0), device=device)
+    y = torch.randn((4, 6), device=device)
+
+    err_msg = "the written-to tensor refer to a single memory location"
+
+    yield ErrorInput(SampleInput((x, y), kwargs={'dim': 0, 'out': x}),
+                     error_regex=err_msg)
+    yield ErrorInput(SampleInput((x, y), kwargs={'dim': 0, 'out': y}),
+                     error_regex=err_msg)
+
+    z = torch.zeros((4, 6), device=device)
+    yield ErrorInput(SampleInput((y, z), kwargs={'out': z[:2, :]}),
+                     error_regex=err_msg)
+
+    # error inputs for different devices
+    if device == 'cuda':
```

Comment:
Check for the device's type being equal to `cuda`: `torch.device(device).type == 'cuda'`

Reply:
Oops, my bad. U